include/swift/Basic/OptionSet.h

```diff
@@ -136,6 +141,13 @@ class OptionSet {
 
   static void _checkResultTypeOperatorOr(...) {}
 
+  static StorageType combineFlags(const std::initializer_list<Flags> &flags) {
+    StorageType storage{};
+    for (Flags flag : flags)
+      storage |= static_cast<StorageType>(flag);
+    return storage;
```

Comment:
WDYT about doing the computation in an OptionSet, and then extracting the storage? That'd be less tricky I think.

```suggestion
    OptionSet<Flags> result;
    for (Flags flag : flags)
      result |= flag;
    return result.storage;
```

Reply:
I like it -- done!